src/Tools/Extensions.ApiDescription.Client/test/GetOpenApiReferenceMetadataTest.cs

```diff
@@ -60,7 +60,9 @@ public void Execute_AddsExpectedMetadata()
             orderedMetadata.Add(key, metadata[key]);
         }
 
-        Assert.Equal(expectedMetadata, orderedMetadata);
+       expectedMetadata["SerializedMetadata"] = string.Join("|", expectedMetadata["SerializedMetadata"].Split('|').OrderBy(s => s));
```

Comment:
It's not clear to me why we need to make this mutation to the `SerializedMetadata` field on S390X targets. Can you provide an example of what the current metadata output looks like? The logs you shared appear to be truncated and don't show the actual comparison of the `SerializedMetadata` field.

Reply:
According to my understanding, This might be due to when serializing metadata https://github.com/dotnet/aspnetcore/blob/20743f2e5ad9a8022d73c15c04d41a67a82e7ac4/src/Tools/Extensions.ApiDescription.Client/src/MetadataSerializer.cs#L62
where CloneCustomMetadata usually returns an `Dictionary` where the order might be undefined.

Here is the difference in this particular test case on s390x
```
Expected: [ClassName, NSwagClient][CodeGenerator, NSwagCSharp][FirstForGenerator, true][Namespace, Console.Client][OriginalItemSpec, TestProjects/files/NSwag.json][OutputPath, obj/NSwagClient.cs][SerializedMetadata, Identity=TestProjects/files/NSwag.json|FirstForGenerator=true|CodeGenerator=NSwagCSharp|OutputPath=obj/NSwagClient.cs|Namespace=Console.Client|OriginalItemSpec=TestProjects/files/NSwag.json|ClassName=NSwagClient]
--------------------------------------------------------------------------------------------
Actual: [ClassName, NSwagClient][CodeGenerator, NSwagCSharp][FirstForGenerator, true][Namespace, Console.Client][OriginalItemSpec, TestProjects/files/NSwag.json][OutputPath, obj/NSwagClient.cs][SerializedMetadata, Identity=TestProjects/files/NSwag.json|FirstForGenerator=true|Namespace=Console.Client|OriginalItemSpec=TestProjects/files/NSwag.json|ClassName=NSwagClient|OutputPath=obj/NSwagClient.cs|CodeGenerator=NSwagCSharp]
```
Note this this change in order not only wrt architecture but with runtime's as well.
here is the change in order for x64 mono
```
expected: [ClassName, NSwagClient][CodeGenerator, NSwagCSharp][FirstForGenerator, true][Namespace, Console.Client][OriginalItemSpec, TestProjects/files/NSwag.json][OutputPath, obj/NSwagClient.cs][SerializedMetadata, Identity=TestProjects/files/NSwag.json|FirstForGenerator=true|CodeGenerator=NSwagCSharp|OutputPath=obj/NSwagClient.cs|Namespace=Console.Client|OriginalItemSpec=TestProjects/files/NSwag.json|ClassName=NSwagClient]
--------------------------------------------------------------------------------------------
Actual: [ClassName, NSwagClient][CodeGenerator, NSwagCSharp][FirstForGenerator, true][Namespace, Console.Client][OriginalItemSpec, TestProjects/files/NSwag.json][OutputPath, obj/NSwagClient.cs][SerializedMetadata, Identity=TestProjects/files/NSwag.json|CodeGenerator=NSwagCSharp|OriginalItemSpec=TestProjects/files/NSwag.json|FirstForGenerator=true|OutputPath=obj/NSwagClient.cs|ClassName=NSwagClient|Namespace=Console.Client]

```

Reply:
@captainsafia can we reopen this PR?